### PR TITLE
Add tests for parsing chunked data split before or in the final CRLF (#4630)

### PR DESCRIPTION
**Pull request by JustAnotherArchivist**
## What do these changes do?

This adds four tests for issue #4630.

## Are there changes in behavior for the user?

Not for users, but for developers: three of the tests are currently failing (see that issue for details).

## Related issue number

#4630

## Checklist

- [ ] *(not applicable)* I think the code is well written
- [x] Unit tests for the changes exist
- [ ] *(not applicable)* Documentation reflects the changes
- [ ] *(not applicable)* If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
- [ ] *(not applicable)* Add a new news fragment into the `CHANGES` folder

### CODE REVIEW

File: tests/test_http_parser.py [hunk 1]
```diff
@@ -826,6 +826,44 @@ class TestParsePayload:
         assert isinstance(out.exception(),
                           http_exceptions.TransferEncodingError)
 
+    async def test_parse_chunked_payload_split_end(self, protocol) -> None:
```
Comment (webknjaz) [inline tests/test_http_parser.py:832]: Please mark all these tests as xfail and add a link to https://github.com/aio-libs/aiohttp/issues/4630 in the reason arg. This way we'll be able to merge failing tests while keeping the CI green.

(ref: https://pganssle-talks.github.io/xfail-lightning)

Comment (JustAnotherArchivist) [reply tests/test_http_parser.py:832]: Ah, I was wondering about that. Thanks, first time working with pytest. :-)

File: tests/test_http_parser.py [hunk 2]
```diff
@@ -826,6 +826,58 @@ class TestParsePayload:
         assert isinstance(out.exception(),
                           http_exceptions.TransferEncodingError)
 
+    @pytest.mark.xfail(
+        raises=AssertionError,
+        reason="see https://github.com/aio-libs/aiohttp/issues/4630"
+    )
```
Comment (webknjaz) [inline tests/test_http_parser.py]: You don't need to specify `raises=AssertionError`. It's usually needed to specify other exceptions that happen, not assertions.
```suggestion
    @pytest.mark.xfail(reason="see https://github.com/aio-libs/aiohttp/issues/4630")
```

File: tests/test_http_parser.py [hunk 3]
```diff
@@ -826,6 +826,58 @@ class TestParsePayload:
...
+        p.feed_data(b'4\r\nasdf\r\n0\r\n')
+        p.feed_data(b'\r\n')
+
+        assert out.is_eof()
+        assert b'asdf' == b''.join(out._buffer)
+
+    @pytest.mark.xfail(
+        raises=AssertionError,
+        reason="see https://github.com/aio-libs/aiohttp/issues/4630"
+    )
```
Comment (webknjaz) [inline tests/test_http_parser.py]:
```suggestion
    @pytest.mark.xfail("see https://github.com/aio-libs/aiohttp/issues/4630")
```

File: tests/test_http_parser.py [hunk 4]
```diff
@@ -826,6 +826,58 @@ class TestParsePayload:
...
+        p.feed_data(b'Content-MD5: 912ec803b2ce49e4a541068d495ab570\r\n')
+        p.feed_data(b'\r\n')
+
+        assert out.is_eof()
+        assert b'asdf' == b''.join(out._buffer)
+
+    @pytest.mark.xfail(
+        raises=AssertionError,
+        reason="see https://github.com/aio-libs/aiohttp/issues/4630"
+    )
```
Comment (webknjaz) [inline tests/test_http_parser.py]:
```suggestion
    @pytest.mark.xfail(reason="see https://github.com/aio-libs/aiohttp/issues/4630")
```

**Pull request merged**